conman/routes/managers.py

```diff
@@ -72,3 +73,16 @@ def move_branch(self, old_url, new_url):
             Value(new_url),
             Substr('url', len(old_url) + 1),  # 1 indexed
         ))
+
+    def with_level(self, level=None):
+        """
+        Annotate the queryset with the (1-indexed) level of each item.
```

Comment:
I see why you chose 1-indexed, but I'm not sure that's the best api. It seems more natural to me to care about how many parts the url has rather than the number of `/` separators.

Reply:
Yeah, I wasn't sure about it myself. I went for this simply because the implementation was marginally easier -- you'll probably not be the last person to think it's strange -- I'll look into changing it.

Reply:
d3963e4